instance/models/openedx_instance.py

```diff
@@ -172,20 +205,38 @@ def get_active_appservers(self):
             return self.ref._cached_active_appservers
         return self.appserver_set.filter(_is_active=True)
 
+    def domain_nginx_regex(self, site_name):
+        """
+        Regex that matches either the internal or external URL for the site.
+
+        This is meant exclusively for filling in the server_name regex in nginx configs.
+        """
+        domain_attrs = ['external_{}_domain', 'internal_{}_domain']
+        domains = [getattr(self, attr.format(site_name), None) for attr in domain_attrs]
+        domains = [domain for domain in domains if domain is not None]
+        choices = '|'.join(map(re.escape, domains))  # pylint: disable=bad-builtin
+        return '~^({})$'.format(choices)
```

Comment:
@haikuginger I believe this `~` does not belong to the regex itself, but to nginx - is there a way to add it somewhere later, so that this method actually returns a working regex.